cmd/runtimetest/main.go

```diff
@@ -630,6 +630,54 @@ func validateMountsExist(spec *rspec.Spec) error {
 	return nil
 }
 
+func validateMountsOrder(spec *rspec.Spec) error {
+	if runtime.GOOS == "windows" {
+		return nil
+	}
+
+	mountInfos, err := mount.GetMounts()
+	if err != nil {
+		return err
+	}
+
+	type mountOrder struct {
+		Order  int
+		Root   string
+		Dest   string
+		Source string
+	}
+	mountsMap := make(map[string][]mountOrder)
+	for i, mountInfo := range mountInfos {
+		m := mountOrder{
+			Order:  i,
+			Root:   mountInfo.Root,
+			Dest:   mountInfo.Mountpoint,
+			Source: mountInfo.Source,
+		}
+		mountsMap[mountInfo.Mountpoint] = append(mountsMap[mountInfo.Mountpoint], m)
+	}
+	current := -1
+	for _, specMount := range spec.Mounts {
+		mounts := mountsMap[specMount.Destination]
+		for _, mount := range mounts {
+			source := mount.Source
+			if specMount.Type == "bind" || specMount.Type == "rbind" {
```

Comment:
(r)bind mounts are still not covered in the spec.  My most recent attempt was opencontainers/runtime-spec#771, which I think should be reopened, but there is also an in-flight opencontainers/runtime-spec#878.  Making runtime-tools decisions based on in-flight spec changes is dicey, but in both of those PRs the bind-ness of a `mounts` entry is based on `options`, not on `type`.  So if you want to land special bind handling at all here, I suggest you look in `specMount.Options`.